myapp/scheduler/stock_api_update.py

```diff
@@ -1,29 +1,26 @@
 from myapp.models import Stock
 from myapp import stock_api
+from django.db import transaction
 
 
-def update_top_stocks():
+@transaction.atomic
+def stock_api_update():
     top_stocks = stock_api._get_top_stocks()
```

Comment:
here two exceptions could happen needs to handle (update your branch with the master to see how I handle exceptions from the service)

Reply:
you mean exception when api called ? to check if the API called fine ?

Reply:
yes, see in `get_stock_info`  what exceptions might happen

Reply:
@murradkh the two exception already defined ! , can you please specify what changes are required ? i'm confused